src/routes/volume/detail/Snapshots.js

```diff
@@ -44,24 +44,34 @@ class Snapshots extends React.Component {
         })
         return
       }
-      let actions
+
       if (action.type === 'snapshotDelete') {
-        actions = [{
+        let additionalActions = [{
           url: this.props.volume.actions.snapshotPurge,
         }]
-      }
-      this.props.dispatch({
-        type: 'snapshotModal/snapshotAction',
-        payload: {
-          type: action.type,
-          actions,
-          url: this.props.volume.actions[action.type],
-          params: {
-            name: action.payload && action.payload.snapshot ? action.payload.snapshot.name : undefined,
+        let actionType = action.type
+        let url = this.props.volume.actions[action.type]
+        let params = {
+          name: action.payload && action.payload.snapshot ? action.payload.snapshot.name : undefined,
+        }
+        // Only snapshotPurge operations are performed when the snapshot status is marked as removed.
+        if (action.payload?.snapshot?.removed) {
+          additionalActions = []
+          actionType = 'snapshotPurge'
+          url = this.props.volume.actions?.snapshotPurge
+          params = {}
+        }
+        this.props.dispatch({
+          type: 'snapshotModal/snapshotAction',
+          payload: {
+            type: actionType,
+            actions: additionalActions,
+            url,
+            params,
+            querySnapShotUrl: this.props.volume.actions.snapshotList,
           },
-          querySnapShotUrl: this.props.volume.actions.snapshotList,
-        },
-      })
+        })
+      }
```

Comment:
```js
	  let actionType = action.type
	  let url = this.props.volume.actions[action.type]
	  let params = {
	    name: action.payload && action.payload.snapshot ? action.payload.snapshot.name : undefined,
	  }

	  if (action.type === 'snapshotDelete') {
		    let additionalActions = [{
		      url: this.props.volume.actions.snapshotPurge,
		    }]
	  
		    // Only snapshotPurge operations are performed when the snapshot status is marked as removed.
		    if (action.payload?.snapshot?.removed) {
		      additionalActions = []
		      actionType = 'snapshotPurge'
		      url = this.props.volume.actions?.snapshotPurge
		      params = {}
		    }
		}
	  
	  this.props.dispatch({
	    type: 'snapshotModal/snapshotAction',
	    payload: {
	      type: actionType,
	      actions: additionalActions,
	      url,
	      params,
	      querySnapShotUrl: this.props.volume.actions.snapshotList,
	    },
	  })
```

Reply:
I just checked again that your comments is correct. I had thought that all other action.types were returned. But I double-checked action.revet , which does execute the default `snapshotModal/snapshotAction`.  Thank you for your patience in checking ！